counter.rb

```diff
@@ -0,0 +1,28 @@
+class Counter
+  def initialize(filename)
+    @filename = filename
+  end
+
+  # Do everything
+  def run
+    contents = File.read(@filename)
+    word_hash = split_text(contents)
+    print_descending(word_hash)
+  end
+
+  # Splits the text into a hash of words and counts
+  def split_text(text)
+    all_words = text.scan(/((?:\w+|\')+)/).flatten.map { |x| x.downcase }
+
+    mapping = Hash.new
```

Comment:
Anytime I see a pattern where we initialize a thing, do something to it, and return it...I always ask myself could I have used `.tap` or `.reduce`.